### PR TITLE
Add GH Action builder

### PR DESCRIPTION
**Pull request by leszko**
@hasancelik after merging this one, I'll remove Jenkins PR Builder job.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+          architecture: ${{ matrix.architecture }}
+
+      - uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-maven-
+
+      - name: Build with Maven
+        run: mvn -B clean verify
```
Comment (pivovarit) [inline .github/workflows/build.yml:33]:
```suggestion
        run: mvn -B verify
```

nitpicking: no need to _clean_

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+name: build
```
Comment (hasancelik) [inline .github/workflows/build.yml:1]: It is very minor but we can use more specific name for the action, I think. What about `PR Builder`? :) @leszko

Comment (leszko) [reply .github/workflows/build.yml:1]: The thing is that AFAIK It's not just for PRs, it also builds master. So, it's quite generic, then the name is also generic `build` ;)

Comment (hasancelik) [reply .github/workflows/build.yml:1]: I see 👍 :)

**Pull request merged**